src/main/java/hudson/plugins/gradle/GradleExecution.java

```diff
@@ -0,0 +1,332 @@
+package hudson.plugins.gradle;
+
+import java.io.IOException;
+import java.nio.charset.Charset;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import org.apache.commons.lang.StringUtils;
+
+import com.google.common.base.Joiner;
+import com.google.common.collect.ImmutableList;
+
+import hudson.EnvVars;
+import hudson.FilePath;
+import hudson.Launcher;
+import hudson.Util;
+import hudson.model.AbstractBuild;
+import hudson.model.Computer;
+import hudson.model.Node;
+import hudson.model.Result;
+import hudson.model.Run;
+import hudson.model.TaskListener;
+import hudson.util.ArgumentListBuilder;
+import hudson.util.VariableResolver;
+
+/**
+ * Helper for execution of an Gradle build.
+ * Encapsulates the real gradle execution independent of build type.
+ */
+public final class GradleExecution {
+
+    private String switches;
```

Comment:
Looks like all these fields can be final?

Reply:
Yes, you are right!
I'll fix.